### PR TITLE
Remove no longer needed ConnectorSplitSource#isFinished

### PR DESCRIPTION
**Pull request by kokosing**
Remove no longer needed ConnectorSplitSource#isFinished

### CODE REVIEW

File: presto-main/src/main/java/io/prestosql/execution/scheduler/SourcePartitionedScheduler.java [hunk 1]
```diff
@@ -317,7 +317,7 @@ else if (pendingSplits.isEmpty()) {
         //     which may contain recently published splits. We must not ignore those.
         //   * If any scheduleGroup is still in DISCOVERING_SPLITS state, it means it hasn't realized that there will be no more splits.
         //     Next time it invokes getNextBatch, it will realize that. However, the invocation will fail we tear down splitSource now.
```
Comment (kokosing) [inline presto-main/src/main/java/io/prestosql/execution/scheduler/SourcePartitionedScheduler.java]: update the comment

File: presto-main/src/main/java/io/prestosql/execution/scheduler/SourcePartitionedScheduler.java [hunk 2]
```diff
@@ -317,7 +317,7 @@ else if (pendingSplits.isEmpty()) {
...
-        if ((state == State.NO_MORE_SPLITS || state == State.FINISHED) || (noMoreScheduleGroups && scheduleGroups.isEmpty() && splitSource.isFinished())) {
+        if ((state == State.NO_MORE_SPLITS || state == State.FINISHED) || (noMoreScheduleGroups && scheduleGroups.isEmpty())) {
```
Comment (sopel39) [inline presto-main/src/main/java/io/prestosql/execution/scheduler/SourcePartitionedScheduler.java]: @dain does this look correct to you?

It seems that `SplitSource#isFinished()` is redundant as split scheduler should drain splits from all partitions/buckets anyway.

File: presto-main/src/main/java/io/prestosql/execution/scheduler/SourcePartitionedScheduler.java [hunk 3]
```diff
@@ -317,7 +317,7 @@ else if (pendingSplits.isEmpty()) {
         //     which may contain recently published splits. We must not ignore those.
         //   * If any scheduleGroup is still in DISCOVERING_SPLITS state, it means it hasn't realized that there will be no more splits.
         //     Next time it invokes getNextBatch, it will realize that. However, the invocation will fail we tear down splitSource now.
-        if ((state == State.NO_MORE_SPLITS || state == State.FINISHED) || (noMoreScheduleGroups && scheduleGroups.isEmpty() && splitSource.isFinished())) {
```
Comment (sopel39) [inline presto-main/src/main/java/io/prestosql/execution/scheduler/SourcePartitionedScheduler.java]: you need to update comment above too

**Comment by kokosing** [comment]
There is some deadlock:
```
2019-08-06T03:58:45.795-0500	WARN	TestHangMonitor	io.prestosql.tests.LogTestDurationListener	No test started or completed in 5.00m. Running tests:
	io.prestosql.plugin.hive.TestHiveIntegrationSmokeTest::testGroupedExecution running for 7.65m
	io.prestosql.plugin.hive.TestHiveIntegrationSmokeTest::testShowColumnMetadata running for 5.03m
	io.prestosql.plugin.hive.TestHiveIntegrationSmokeTest running for 21.21m.
```

**Comment by colebow** [comment]
:wave: @kokosing - this PR is inactive and doesn't seem to be under development, and it might already be implemented. If you'd like to continue work on this at any point in the future, feel free to re-open.

**Pull request closed without merging**